vllm/model_executor/layers/linear.py

```diff
@@ -623,8 +626,24 @@ def weight_loader_v2(self,
         assert loaded_shard_id < len(self.output_sizes)
 
         tp_size = get_tensor_model_parallel_world_size()
-        shard_offset = sum(self.output_sizes[:loaded_shard_id]) // tp_size
-        shard_size = self.output_sizes[loaded_shard_id] // tp_size
+
+        if isinstance(param, BlockQuantScaleParameter):
```

Comment:
Do you know why this is needed?

Also, do you know why it is only needed for `MergedColumnParallel` and not `QKVParallel`